outlines/prompts.py

```diff
@@ -118,10 +125,14 @@ def _template_from_file(_, path: Path) -> jinja2.Template:
             keep_trailing_newline=True,
             undefined=jinja2.StrictUndefined,
         )
+
+        for name, filter_fn in filters.items():
```

Comment:
I generally don't mind a little repetition but I think we should refactor the code and add a `create_environment` class that can be used in both class methods. @yvan-sraka are there any reasons we *shouldn't* do that?

Reply:
I agree, ideally, it would be less surprising for users to have the most similar environment (and behavior) for any method used to create a `Prompt`. Otherwise, it's a great feature addition!